middleware/auth.js

```diff
@@ -0,0 +1,30 @@
+import admin from 'firebase-admin';
+
+const auth = function (req, res, next) {
+  // Get token from header
+  const token = req.header('x-auth-token');
+
+  // Check if not token
+  if (!token) {
+    return res.status(401).json({ msg: 'No token, authorization denied' });
+  }
+
+  try {
+    // verify firebase token
+    admin
+      .auth()
+      .verifyIdToken(token)
+      .then(decodedToken => {
+        req.user = decodedToken; // to get id use decodedToken.uid
+        next();
+        // res.status(200).send(decodedToken);
+      })
+      .catch(() => {
+        res.status(403).send('Unauthorized');
+      });
+  } catch (err) {
+    res.status(401).json({ msg: 'Token is not valid' });
```

Comment:
I'm not sure if this line will be reached with the promise catch

Reply:
I'll run some tests 👍

Reply:
This gets hit if we forget to include `admin.initializeApp()` for firebase, can be useful for debugging as i received this when i forgot to initialise firebase on the backend.

Reply:
Then we are printing for that page that the token is not valid when the error comes from firebase initialization. 
We should check if the firebase has already been initialized before launching the server with this command: `admin.apps.length !== 0`. Otherwise, this error will be printed over and over.